lib/rbac/filterer.rb

```diff
@@ -252,12 +257,36 @@ def search(options = {})
               .includes(include_for_find).includes(exp_includes)
               .order(order)
 
-      scope = include_references(scope, klass, include_for_find, exp_includes, skip_references)
       scope = scope.limit(limit).offset(offset) if attrs[:apply_limit_in_sql]
-      targets = scope
```

Comment:
This ties into my comment about avoiding re-using the scope variable below, but can this deletion also be avoided and left until after the `if !options[:use_sql_view]` block?

Inside the end of the `else`, you might need to do a `scope.length`, but that is part of the changed bit anyway, so I am less concerned about that having a bit of inconsistency for now.